Core: Add setting to hide the TaskWatcher

theo-vt opened the pull request:
<!-- Include a brief summary of the changes. -->

The TaskWatcher is a great didactic tool but it can clutter the view, especially on smaller screen, this PR adds a setting to hide the TaskWatcher

<!--
The FreeCAD community thanks you for your contribution!
By creating a Pull Request you agree to the contributing policy. The complete policy can be found in the root of the source tree (CONTRIBUTING.md) or at https://github.com/FreeCAD/FreeCAD/blob/main/CONTRIBUTING.md

This template provides guidance on creating a PR that can be reviewed and approved as quickly as possible. Comments may be safely deleted.

Unless you know exactly what you're doing, please leave the checkbox 'Allow edits by maintainers' enabled.  This will allow maintainers to help you.
-->

<!-- link to individual issues this PR closes by referencing the issue number (e.g., fixes #1234, closes #4321). -->

## Before and After Images
<!-- If your proposed changes affect the FreeCAD GUI, add before and after screenshots -->
<img width="1917" height="757" alt="Screenshot From 2025-07-31 12-02-40" src="https://github.com/user-attachments/assets/4a202970-ac2a-4b58-8ed1-c2cb55eb7d32" />
<img width="1920" height="679" alt="Screenshot From 2025-07-31 12-11-10" src="https://github.com/user-attachments/assets/a39156e5-ef8d-46be-8a07-3ac3bce7eadf" />

with the new setting located under General/Selection

<img width="964" height="678" alt="Screenshot From 2025-07-31 12-04-41" src="https://github.com/user-attachments/assets/39e54c79-2952-4f89-a464-6d3d6300be6e" />



<!--  Notes on the PR Review Process

The following section describes what the maintainers consider when reviewing your Pull Request.  These items may not require you to take any action.  This information is provided for context. Understanding what we consider will help you prepare your request for speedy approval.

You can find additional documentation about these guidelines in the [Developers handbook](https://freecad.github.io/DevelopersHandbook).

Alignment (Does the PR align with the goals and interests of the project?)
  - Does the PR have at least one issue linked, which this PR closes?
  - Has the conversation on the PR and related issue(s) reached consensus?
  - If the PR affects the GUI, is the Design Working Group (DWG) aware and have they had time to review and comment?
  - If the PR affects the GUI, did the contributor include before/after images?
  - If the PR affects standards and workflow, is the CAD Working Group (CWG) aware and have they had time to review/comment?

Impact (Does the change affect other parts of the project?)
  - Has the impact on documentation been considered and appropriate action taken?
  - Has the impact on translation been considered appropriate action taken?
  - Will the PR affect existing user documents?

Code Quality (Is code well-written and maintainable?)
  - Does the PR warrant a review by the Code Quality Working Group (CQWG)?
  - Does the change include tests?
  - Is the PR rebased on the current main branch with unnecessary commits squashed?

Release (Are there considerations related to release timing?)
  - Has the PR been considered for backporting to the latest release branch?
  - Have the release notes been considered/updated?
  -->

theo-vt commented:
Maybe Display/UI and add the TaskWatcher box?

kadet1090 commented:
UI seems like a good place, but it should be named something like `Suggested Actions`. Task Watcher is highly technical term coming straight from code and has no practical meaning to users.

adrianinsaval commented:
This feature is activated on selection so I think it does make some sense to put it in selection

theo-vt commented:
I moved the setting and changed the wording:

<img width="961" height="900" alt="image" src="https://github.com/user-attachments/assets/3a8f8fe1-bd0c-4a9a-8c67-72add141b179" />

theo-vt commented on src/Gui/PreferencePages/DlgSettingsUI.ui:
```diff
@@ -497,10 +497,32 @@
-      <enum>Qt::Orientation::Vertical</enum>
+    <widget class="QGroupBox" name="groupBox_4">
+     <property name="title">
+      <string>Suggested actions</string>
```
Wouldn't that go against the rest of the page?

The other titles don't use sentence case, ("Tree view", "Theme customization")

maxwxyz replied:
The rest of the page is Title Case too: 
<img width="1782" height="1596" alt="grafik" src="https://github.com/user-attachments/assets/7f6a1933-0750-4f86-886e-cec8c1479235" />
